pkg/grafana/client.go

```diff
@@ -0,0 +1,137 @@
+package grafana
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/kanopy-platform/k8s-auth-portal/pkg/random"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	ErrRoleNotValid = errors.New("role is not valid")
+)
+
+type Client struct {
+	client GAPIClient
+}
+
+type GAPIClient interface {
+	UserByEmail(email string) (user gapi.User, err error)
+	CreateUser(user gapi.User) (int64, error)
+	AddOrgUser(orgID int64, user, role string) error
+	UpdateOrgUser(orgID, userID int64, role string) error
+}
+
+func NewClient(baseURL *url.URL, cfg gapi.Config) (*Client, error) {
+	newClient := &Client{}
+
+	client, err := gapi.New(baseURL.String(), cfg)
```

Comment:
You probably want to set https://github.com/grafana/grafana-api-golang-client/blob/9bd3cddb7c2b00364d2c55e39c3db7640558c6b7/client.go#L23 to a reasonably configured *http.Client with an explicit timeout so that we pass an error from the proxy back to the client and not a 503 from the load balancer.

Reply:
- What would be sensible values? I noticed that in gapi they use https://github.com/hashicorp/go-cleanhttp/blob/master/cleanhttp.go, that it seems to contain those values.
- Should NewClient provide those default values or set default on the cli and make some configurable with extra parameters?

Reply:
Huh, I missed that they are already setting defaults in their client constructor. These might be fine I just wanted some timeout vs wait forever.

Reply:
If you wanted to tune them, your proxy has < 60 seconds to complete the entire login handshake and have grafana respond with a session. These defaults are too long for the timeout response to be returned before the ELB hangs up so tuning them to be short enough that the entire flow, including the response from the final request to grafana, is ensured to fit into 60s would be the desired defaults. That might be a premature optimization though.